test/test_intensify/test_intensify.py

```diff
@@ -398,40 +398,145 @@ def target(x):
-            instances=[1], run_obj_time=False,
+            instances=[1, 2], run_obj_time=False,
             deterministic=False, always_race_against=self.config3, run_limit=1)
 
+        self.assertEqual(intensifier.n_iters, 0)
+
+        # intensification iteration #1
         # run incumbent first if it was not run before
-        config, _ = intensifier.get_next_challenger(challengers=[self.config2, self.config1, self.config3],
+        config, _ = intensifier.get_next_challenger(challengers=[self.config2, self.config1],
                                                     chooser=None)
-        inc, _ = intensifier.eval_challenger(challenger=config, incumbent=None,
-                                             run_history=self.rh,)
-
+        self.assertEqual(config, self.config2)
+        self.assertEqual(intensifier.stage, IntensifierStage.RUN_INCUMBENT)
+        # eval config 2 (=first run)
+        inc, _ = intensifier.eval_challenger(challenger=config, incumbent=None, run_history=self.rh, )
         self.assertEqual(inc, self.config2)
         self.assertEqual(intensifier.stage, IntensifierStage.RUN_CHALLENGER)
+        self.assertEqual(self.stats.inc_changed, 1)
 
         # run challenger now that the incumbent has been executed
-        config, _ = intensifier.get_next_challenger(challengers=[self.config2, self.config1, self.config3],
+        config, _ = intensifier.get_next_challenger(challengers=None,  # don't need a new list here as old one is cont'd
                                                     chooser=None)
-        inc, _ = intensifier.eval_challenger(challenger=config, incumbent=inc,
-                                             run_history=self.rh,)
+        self.assertEqual(intensifier.stage, IntensifierStage.RUN_CHALLENGER)
+        self.assertEqual(config, self.config1)
+        inc, _ = intensifier.eval_challenger(challenger=config, incumbent=inc, run_history=self.rh, )
 
-        # challenger should have a better performance, so incumbent should have changed
+        # challenger has a better performance, so incumbent has changed
         self.assertEqual(inc, self.config1)
-        self.assertEqual(self.stats.inc_changed, 1)
+        self.assertEqual(self.stats.inc_changed, 2)
         self.assertEqual(intensifier.stage, IntensifierStage.RUN_DEFAULT)
         self.assertFalse(intensifier.continue_challenger)
 
-        # run `always_race_against` now since the incumbent has changed
-        config, _ = intensifier.get_next_challenger(challengers=[self.config2, self.config1, self.config3],
+        # run `always_race_against`
+        config, _ = intensifier.get_next_challenger(challengers=None,  # don't need a new list here as old one is cont'd
                                                     chooser=None)
-        inc, _ = intensifier.eval_challenger(challenger=config, incumbent=inc,
-                                             run_history=self.rh,)
+        self.assertEqual(config, self.config1)
```

Comment:
Would it be possible to return config3 here as this is the config that will be evaluated by eval_challenger?